bootstrap/bootstrap.py

```diff
@@ -334,55 +320,146 @@ def make_archive(cabal_path):
 
     return archivename
 
+def fetch_from_plan(plan : FetchPlan, output_dir : Path):
+  import urllib.request
```

Comment:
Move this to the top level, start of the file? (I see it was inlined before, but I think it's not good to hide possible side effects of importing a module inside a function body.)

Also the indentation doesn't match the rest of the file